### PR TITLE
feat: create lint tests

### LINKED ISSUE #11: [FR]: Expose lint actions as tests
### What is the current behavior?

You have to use `--keep_going` if you want a bazel build to print all the lint violations and also exit non-zero.

### Describe the feature

A lot of users wire up linters today as tests, so we should let them continue to do that.

### PR DESCRIPTION
**Pull request by alexeagle**
Fixes #11

FYI @bencodes

Demo:
```
example % bazel test test:pmd   
INFO: Analyzed target //test:pmd (9 packages loaded, 990 targets configured).
INFO: Found 1 test target...
INFO: From PMD src/foo.PMD-report.txt:
Oct 14, 2023 8:18:06 AM net.sourceforge.pmd.PMD encourageToUseIncrementalAnalysis
WARNING: This analysis could be faster, please consider using Incremental Analysis: https://pmd.github.io/pmd-6.55.0/pmd_userdocs_incremental_analysis.html
FAIL: //test:pmd (see /private/var/tmp/_bazel_alexeagle/3eecd93732ac7c6f4f360a54a468da50/execroot/_main/bazel-out/darwin_arm64-fastbuild/testlogs/test/pmd/test.log)
INFO: From Testing //test:pmd:
==================== Test output for //test:pmd:
src/Foo.java:8: FinalizeOverloaded:     Finalize methods should not be overloaded
================================================================================
Target //test:pmd up-to-date:
  bazel-bin/test/assert_no_lint_warnings.sh
INFO: Elapsed time: 13.232s, Critical Path: 2.97s
INFO: 14 processes: 8 internal, 5 darwin-sandbox, 1 worker.
//test:pmd                                                               FAILED in 0.3s
  /private/var/tmp/_bazel_alexeagle/3eecd93732ac7c6f4f360a54a468da50/execroot/_main/bazel-out/darwin_arm64-fastbuild/testlogs/test/pmd/test.log

Executed 1 out of 1 test: 1 fails locally.
```

### CODE REVIEW

File: lint/assert_no_lint_warnings.bzl [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        output = bin,
+        substitutions = {"{{reports}}": " ".join([to_rlocation_path(ctx, r) for r in reports])},
+        is_executable = True,
+    )
+    return [DefaultInfo(
+        executable = bin,
+        runfiles = ctx.runfiles(reports + [ctx.file._runfiles_lib]),
+    )]
+
+def assert_no_lint_warnings(aspect):
```
Comment (gregmagolan) [inline lint/assert_no_lint_warnings.bzl:58]: how about `make_lint_test`?

usage then reads,

```
flake8_test = make_lint_test(aspect = flake8)
```

File: lint/assert_no_lint_warnings.bzl [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    return rule(
+        implementation = _test_impl,
+        attrs = {
+            "srcs": attr.label_list(doc = "*_library targets", aspects = [aspect]),
+            "fail_on_violation": attr.bool(),
```
Comment (gregmagolan) [inline lint/assert_no_lint_warnings.bzl:63]: fail_on_violation looks unused

Comment (alexeagle) [reply lint/assert_no_lint_warnings.bzl]: commented - it's required

File: lint/assert_no_lint_warnings.bzl (hunk 1 above)
Comment (alexeagle) [reply lint/assert_no_lint_warnings.bzl]: yeah that's nicer

**Review by gregmagolan — APPROVED** [review]
🌯

**Pull request merged**